### PR TITLE
Ghost translation

### PR DESCRIPTION
**Pull request by markusbattarbee**
Replaces #609 
Incorporates #1000

This PR introduces ghost translation (previously known as local translation). Allows for running all Vlasov translation with coalesced MPI just before the actual call, reducing the overhead related to 3D imbalances. This comes at the cost of having to actually compute translation for a boundary of ghost cells. This also requries a grid neighborhood of VLASOV_STENCIL_WIDTH+1 which does incur some overhead.

Turn on ghost translation with `vlasovsolver.GhostTranslate` set to true. This current form of the PR forces this on so we get CI with it active, but it should probably be turned off as default.

The correctness of this method was validated with local tests but this PR should re-confirm it with CI. If so preferred, a smaller ghost neighbourhood can be translated instead of all required, resulting in some diffs at process boundaries, but less of an overhead. This might be useful for example for early flushing of a simulation. The size of the stencil to include in ghost translation can be set with`vlasovsolver.GhostTranslateExtent`. The parameter has a default of `0` which is automatically replaced with `VLASOV_STENCIL_WIDTH+1`. Values greater than that are capped. Thus, possible values are between `1` and `VLASOV_STENCIL_WIDTH+1`.

### CODE REVIEW

**Comment by markusbattarbee** [comment]
WIP (e.g. the default handling breaks the ionosphere tests, I see) and evaluating performance bottlenecks.

File: grid.h [hunk 1]
```diff
@@ -54,7 +54,13 @@ void initializeGrids(
 
     \param[in,out] mpiGrid The DCCRG grid with spatial cells
 */
-void balanceLoad(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid, SysBoundary& sysBoundaries);
+void balanceLoad(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid, SysBoundary& sysBoundaries,
+   bool doTranslationLists = true);
+
+/* helper for calculating AMR flags and cell lists and building pencils
+ */
+void prepareAMRLists(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid);
```
Comment (ykempf) [inline grid.h:62]: Can go if not called outside grid.cpp?

File: parameters.h [hunk 2]
```diff
@@ -60,6 +60,8 @@ struct Parameters {
                                         timestep if useCFLlimit is true. */
    static Real vlasovSolverMinCFL;   /*!< The minimum CFL limit for propagation of distribution function. Used to set
                                         timestep if useCFLlimit is true. */
+   static bool vlasovSolverGhostTranslate;   /*!< Flag for activating all-local ghost translation. */
+   static uint vlasovSolverGhostTranslateExtent;   /*!< Defule stencil size in all-local ghost translation. */
```
Comment (ykempf) [inline parameters.h:64]: define?

File: grid.h (hunk 1 above)
Comment (markusbattarbee) [reply grid.h:62]: Is actually called in grid.cpp before the function is defined, so is needed.

Comment (lkotipal) [reply grid.h:62]: It can be forward declared in `grid.cpp` to reduce scope, but I'm not entirely sure either way is strictly better.

**Comment by ursg** [comment]
#1060 merges this with current dev, to hopefully get rid of the merge conflicts.

**Pull request merged**